README.md

```diff
@@ -40,55 +40,87 @@ In your service worker you just need to use `importScripts` to load Shed
 });
 
-// The built-in handlers are networkFirst, networkOnly, cacheFirst, cacheOnly
-// and fastest. The networkFirst, cacheFirst and fastest handlers will update
-// the cached version if a response is fetched from the network.
-shed.router.get('/myapp/index.html', shed.networkFirst); // Try the network, fallback to cache
-shed.router.get('/myapp/index.html', shed.networkOnly); // Try the network, fail if not available
-shed.router.get('/myapp/index.html', shed.cacheFirst); // Try the cache, fallback to network if not in the cache
-shed.router.get('/myapp/index.html', shed.cacheOnly); // Try the cache, fail if not cached
-shed.router.get('/myapp/index.html', shed.fastest); // Request from the cache and the network, return which ever comes back first
+// For requests to other origins, specify the origin as an option
+shed.router.post('/(.*)', apiHandler, {origin: 'https://api.example.com'});
 
-// You can use relative URLs, which are relative to the scope of the worker
-shed.router.get(':foo/:bar', shed.networkOnly);
+// Provide a default handler
+shed.router.default = myDefaultRequestHandler;
 
-// For requests to other origins, pass the origin in the parameters to the route
-shed.router.post('/(.*)', shed.networkFirst, {origin: 'https://api.example.com'});
+// You can provide a list of resources which will be cached at service worker install time
+shed.precache(['/index.html', '/site.css', '/images/logo.png']);
+```
 
-// Origins can be specified as a RegExp
-shed.router.post('/(.*)', shed.networkFirst, {origin: /https:\/\/.*\.example\.com/});
+## Request handlers
+A request handler receives three arguments
 
-// At the moment, if the fetch is for a URL that doesn't match a route, you get
-// some default behaviour. The 'default default' is networkOnly, but you can
-// change it like so:
-shed.router.default = shed.cacheFirst;
+```javascript
+var myHandler = function(request, values, options) {
+	// ...
+}
+```
 
-// If you want some resources to be cached during the install event, specify
-// them with the precache method
-shed.precache(['/index.html', '/site.css', '/images/logo.png']);
+- `request` - Request object that triggered the `fetch` event
+- `values` - Object whose keys are the placeholder names in the URL pattern, with the values being the corresponding part of the request URL. For example, with a URL pattern of `'/images/:size/:name.jpg'` and an actual URL of `'/images/large/unicorns.jpg'`, `values` would be `{size: 'large', name: 'unicorns'}`
+- `options` - the options object that was used when creating the route (see the API docs below)
 
-// You can manually add or remove things from the cache
-shed.cache('/data/2014/posts.json');
-shed.uncache('/data/2013/posts.json');
+The return value should be a Response, or a promise for a Response. If another value is returned, or if the returned promise is rejected, the Request will fail.
```

Comment:
Mentioned that when the `Request` fails, it will look like a [`NetworkError`](https://developer.mozilla.org/en-US/docs/Web/API/DOMException#exception-NetworkError) from the perspective of the controlled page.